File: ios/Plugin/WalletConnectCapacitorPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(WalletConnectCapacitorPlugin, "WalletConnectCapacitor",
            CAP_PLUGIN_METHOD(connect, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(getNetworkId, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(sign, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(sendTransaction, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(sendCustomRequest, CAPPluginReturnPromise);
)

File: android/src/main/java/com/polepie/walletconnect/capacitor/WalletConnectCapacitorPlugin.java
package com.polepie.walletconnect.capacitor;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "WalletConnectCapacitor")
public class WalletConnectCapacitorPlugin extends Plugin {

    private WalletConnectCapacitor implementation = new WalletConnectCapacitor();

    @PluginMethod
    public void connect(PluginCall call) {
       call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void sign(PluginCall call) {
       call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void getNetworkId(PluginCall call) {
       call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void sendTransaction(PluginCall call) {
       call.unimplemented("Not implemented on Android.");
    }

    @PluginMethod
    public void sendCustomRequest(PluginCall call) {
       call.unimplemented("Not implemented on Android.");
    }

}
